src/agentscope/models/gemini_model.py

```diff
@@ -0,0 +1,293 @@
+# -*- coding: utf-8 -*-
+"""Google Gemini model wrapper."""
+import os
+from collections.abc import Iterable
+from typing import Sequence, Union, Any
+
+from loguru import logger
+
+from agentscope.message import Msg
+from agentscope.models import ModelWrapperBase, ModelResponse
+from agentscope.utils import QuotaExceededError, MonitorFactory
+
+try:
+    import google.generativeai as genai
+except ImportError:
+    genai = None
+
+
+class GeminiWrapperBase(ModelWrapperBase):
+    """The base class for Google Gemini model wrapper."""
+
+    _generation_method = None
```

Comment:
Seems only for Gemini?  Leave a comment about how it will be used?

Reply:
This argument is only for gemini api, which aims at filtering different models in `list_models` function.

Reply:
More explanation is added.